newrelic/hooks/mlmodel_openai.py

```diff
@@ -704,7 +693,9 @@ def wrap_base_client_process_response_sync(wrapped, instance, args, kwargs):
     if not settings.ai_monitoring.enabled:
         return wrapped(*args, **kwargs)
 
-    nr_response_headers = bind_base_client_process_response(*args, **kwargs)
+    bound_args = bind_args(wrapped, args, kwargs)
+    nr_response_headers = getattr(bound_args["response"], "headers", None) or {}
```

Comment:
Do we want to use a `.get()` here for the `response` key?

Reply:
The only time response wouldn't exist is if the function signature changed in which case I think it would probably be better to fail fast and alert us to that change rather than silently passing the issue along to later code.